src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrencePattern.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.FeatureManagement.FeatureFilters
+{
+    /// <summary>
+    /// The recurrence pattern describes the frequency by which the time window repeats.
+    /// </summary>
+    public class RecurrencePattern
+    {
+        /// <summary>
+        /// The recurrence pattern type.
+        /// </summary>
+        public RecurrencePatternType Type { get; set; }
+
+        /// <summary>
+        /// The number of units between occurrences, where units can be in days, weeks, months, or years, depending on the pattern type.
+        /// </summary>
+        public int Interval { get; set; } = 1;
+
+        /// <summary>
+        /// The days of the week on which the time window occurs.
+        /// </summary>
+        public IEnumerable<DayOfWeek> DaysOfWeek { get; set; }
+
+        /// <summary>
+        /// The first day of the week.
+        /// </summary>
+        public DayOfWeek FirstDayOfWeek { get; set; }
```

Comment:
Since this is an existing enum, does the default value work?

Reply:
Yes. The default value of DayOfWeek is Sunday.
``` C#
namespace System
{
    //
    // Summary:
    //     Specifies the day of the week.
    public enum DayOfWeek
    {
        //
        // Summary:
        //     Indicates Sunday.
        Sunday = 0,
        //
        // Summary:
        //     Indicates Monday.
        Monday = 1,
        //
        // Summary:
        //     Indicates Tuesday.
        Tuesday = 2,
        //
        // Summary:
        //     Indicates Wednesday.
        Wednesday = 3,
        //
        // Summary:
        //     Indicates Thursday.
        Thursday = 4,
        //
        // Summary:
        //     Indicates Friday.
        Friday = 5,
        //
        // Summary:
        //     Indicates Saturday.
        Saturday = 6
    }
}
```